src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/InteropServices/PreferIsExactSpellingIsTrueForKnownApis.cs

```diff
@@ -0,0 +1,130 @@
+                                                                             isEnabledByDefaultInFxCopAnalyzers: true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(DefaultRule, WideRule);
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.EnableConcurrentExecution();
+            analysisContext.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            analysisContext.RegisterSymbolAction(AnalyzeExternMethod, SymbolKind.Method);
+        }
+
+        private static void AnalyzeExternMethod(SymbolAnalysisContext context)
+        {
+            var methodSymbol = (IMethodSymbol)context.Symbol;
+            if (!methodSymbol.IsExtern)
+                return;
+            var dllImportAttribute = methodSymbol
+                .GetAttributes()
+                .FirstOrDefault(x => x.AttributeClass.Name.Equals("DllImportAttribute", StringComparison.Ordinal));
+            if (dllImportAttribute is null)
+                return;
+
+            var hasEntryPointParameter = dllImportAttribute.NamedArguments.FirstOrDefault(x => x.Key.Equals("EntryPoint", StringComparison.Ordinal));
+            var hasExactSpellingParameter = dllImportAttribute.NamedArguments.FirstOrDefault(x => x.Key.Equals("ExactSpelling", StringComparison.Ordinal));
+            var hasCharSetParameter = dllImportAttribute.NamedArguments.FirstOrDefault(x => x.Key.Equals("CharSet", StringComparison.Ordinal));
+
+            var methodName = hasEntryPointParameter.Key is null ? methodSymbol.Name : hasEntryPointParameter.Value.Value.ToString();
+            var isExactSpelling = !(hasExactSpellingParameter.Key is null) && (bool)hasExactSpellingParameter.Value.Value;
+            var isCharSetUnicode = !(hasCharSetParameter.Key is null) && ((CharSet)hasCharSetParameter.Value.Value == CharSet.Unicode);
+            var dllName = dllImportAttribute.ConstructorArguments.FirstOrDefault().Value?.ToString() ?? string.Empty;
+            if (KnownApis.Value.TryGetValue(dllName, out var methods))
+            {
+                if (methods.Contains(methodName)) //Whatever exactspelling is is wrong, wide method should be preferred
+                {
+                    if (isCharSetUnicode && !isExactSpelling && methodName.EndsWith("W", StringComparison.Ordinal))
+                    {
+                        var diagnostic = Diagnostic.Create(WideRule, methodSymbol.Locations[0], methodName);
+                        context.ReportDiagnostic(diagnostic);
+                        return;
+                    }
+                    if (!isCharSetUnicode && !isExactSpelling && methodName.EndsWith("A", StringComparison.Ordinal))
+                    {
+                        var diagnostic = Diagnostic.Create(DefaultRule, methodSymbol.Locations[0], methodName);
```

Comment:
Thanks for that. The extensions really are helpful. It would be wounderful if some of this helpfull stuff would also be in the visual studio template! 👍